src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1365,6 +1365,23 @@ export class ResultsViewPageStore {
         }
     });
 
+    readonly genesetLinkMap = remoteData<{[genesetId: string]: string}>({
+        invoke: async () => {
+            if (this.genesetIds && this.genesetIds.length) {
+                return (await internalClient.fetchGenesetsUsingPOST(
+                    {genesetIds: this.genesetIds.slice()}
+                )).reduce(
+                    (linkMap, {genesetId, refLink}) => (
+                        _.assign(linkMap, {[genesetId]: refLink})
```

Comment:
I see you're doing this for a one-line-return, and it is nice and clever :) but not very readable, in my opinion.. can you just expand this all into a for-loop with more common syntax?

Reply:
I'm mostly doing this because I find the reduce more readable as an explicit transformation of an array to an object of a different shape, but I do agree that it would at the very least be better to name the intermediate array here.

Reply:
what is slightly non-readable for me is _.assign(linkMap, ...) instead of linkMap[genesetId] = refLink

Reply:
Right that's a Lodash fill-in for Object.assign(), a combination of that and returning the mutated linkMap to the next iteration. I can easily split those two things over two lines!

Reply:
thanks :)

Reply:
Even better solution that fixes both our objections (mine being pretty awkward to read in a language that doesn't properly understand immutability and yours talking about array indexes in logic that isn't about array indexes) — `Array.prototype.forEach()` to the rescue 😄

Reply:
good point and good solution